tests/test_models.py

```diff
@@ -194,6 +196,26 @@ def test_testrun_status(self):
         status = self.testrun.summary()
         self.assertEqual(1, status.tests_fail)
 
+    def test_testrun_attachment(self):
+        """Test that created attachment entries can be retrieved from the TestRun."""
+        self.assertTrue(self.testrun_attachment.attachments is not None)
+        self.assertTrue(len(self.testrun_attachment.attachments) == 2)
+
+    def test_testrun_attachment_urls(self):
+        """Test that created attachment entries can be accessed via the attachment_urls TestRun property."""
+        attachment_urls = self.testrun_attachment.attachment_urls
+        self.assertTrue(attachment_urls is not None)
+        self.assertTrue(len(attachment_urls) == 2)
+
+        attachment_contents = []
+        expected_attachment_contents = [b'attachment file 1 content', b'attachment file 2 content']
+
+        for attachment_url in attachment_urls:
+            request = requests.get(attachment_url)
```

Comment:
I think we can create an `Attachment` class with a `read` function within so we could do things like:
 
```python
# core/models.py

class Attachment(SquadObject):
    def __init__(self, url):
        self.url = url

    def read(self):
        response = SquadApi.get(self.url)
        # check response.status_code here
        return response.contents

class TestRun(SquadObject):
...
    @attachments.setter
    def attachments(self, attachments):
        if attachments:
            self.__attachments__ = []
            for a in attachments:
                self.__attachments__.append(Attachment(a["download_url"]))
...

# tests/test_models.py

...

contents = []
for a in testrun.attachments:
  contents.append(a.read())

...

```

This way we still centralize all HTTP calls to SquadApi class, which already handle retries and cookies and auth tokens.

And I just realized now that attachments in SQUAD are public. Let's chat about that too tomorrow.